tests/wpunit/TEC/Common/Editor/Full_Site/Template_UtilsTest.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+namespace TEC\Common\Editor\Full_Site;
+
+use InvalidArgumentException;
+use WP_Block_Template;
+
+class Template_UtilsTest extends \Codeception\TestCase\WPTestCase {
+
+
+	/**
+	 * Should create a post and return the template.
+	 *
+	 * @test
+	 */
+	public function should_create_block_template_post() {
+		$faux_post = [
+			'post_name'    => 'bob',
+			'tax_input'    => [ 'wp_theme' => 'tec' ],
+			'post_content' => 'Lorem ipsum...',
+			'post_type'    => 'wp_template',
+		];
+		$id        = static::factory()->user->create( [ 'role' => 'administrator' ] );
+		// Needs access to add term
+		wp_set_current_user( $id );
+		$templateA = Template_Utils::save_block_template( $faux_post );
+		$templateB = Template_Utils::save_block_template( $faux_post );
+		$this->assertInstanceOf( WP_Block_Template::class, $templateA );
+		$this->assertInstanceOf( WP_Block_Template::class, $templateB );
+		$this->assertGreaterThan( 0, $templateA->wp_id );
+		$this->assertGreaterThan( 0, $templateB->wp_id );
+		$this->assertNotEquals( $templateA->wp_id, $templateB->wp_id );
+		wp_set_current_user( 0 );
```

Comment:
The user is reset by the test case after the test, there should be no need to do this.